### PR TITLE
Remove Negative Damage Weapon Full Health Check

### PR DESCRIPTION
**Pull request by MustaphaTR**
The behaviour can be replicated with conditions and currently non-damage warheads on the weapon can prevent this check from properly working.

We talked on irc that some warheads like CreateEffect shouldn't effect the targeting regardless, but i haven't done it yet because couldn't figure how exactly to do it. It can be done in a seperate PR.

I also put an Update rule that notifies about this change, but doesn't actually edit anything.

### CODE REVIEW

File: mods/ra/rules/defaults.yaml
Comment (pchote) [inline mods/ra/rules/defaults.yaml]: We support multiple active `Targetable` traits, right?
IMO it would be cleaner to keep this targetable active all the time, and then make the other one only add `Repair` when needed.

File: mods/ra/rules/defaults.yaml
Comment (pchote) [inline mods/ra/rules/defaults.yaml]: IMO it would be clearer if we could drop the double negatives by defining these as `damaged` conditions, with `ValidDamageStates: Light, Medium, Heavy, Critical`

File: mods/ts/rules/defaults.yaml
Comment (pchote) [inline mods/ts/rules/defaults.yaml]: Need to remove this `Heal`

File: mods/ts/rules/defaults.yaml
Comment (pchote) [inline mods/ts/rules/defaults.yaml]: Style nit: `RequiresCondition` should be the first field on a definition, so keep this where it was.

File: mods/ra/rules/defaults.yaml
Comment (pchote) [inline mods/ra/rules/defaults.yaml]: Style nit: Please move all these `RequiresCondition`s to the top of their definitions.

**Comment by pchote** [comment]
Also needs a rebase.

**Review by pchote — CHANGES_REQUESTED** [review]
RA mechanics can no longer repair vehicles, and I don't see an immediately obvious reason why.

File: mods/ra/rules/defaults.yaml
Comment (abcdefg30) [inline mods/ra/rules/defaults.yaml]: `Repair`, please.

File: mods/ra/rules/defaults.yaml
Comment (pchote) [inline mods/ra/rules/defaults.yaml]: IMO it makes the rules much easier to read if the fields that determine whether a trait is active (`RequiresCondition`, `RequiresPrerequisites`, `Faction` – and in this case `ValidDamageStates`) go at the top, before fields that define what the trait does when it is active (in this case grant `Condition`).

I don't think this needs to block merging here, but it is something I would like to start applying to our default mods.  We should consider adding a section to our coding standards with this and other conventions (e.g. for the trait `@SUFFIX`es and condition names) if we can agree on what the standards should be.

**Review by pchote — DISMISSED** [review]

File: OpenRA.Mods.Common/UpdateRules/Rules/20180923/RemoveNegativeDamageFullHealthCheck.cs
Comment (abcdefg30) [inline OpenRA.Mods.Common/UpdateRules/Rules/20180923/RemoveNegativeDamageFullHealthCheck.cs]: This sentence doesn't make sense to me. `by granting a condition ...`?
(The one below has issues as well.)

**Review by abcdefg30 — DISMISSED** [review]
Looks good otherwise.

File: OpenRA.Mods.Common/UpdateRules/Rules/20180923/RemoveNegativeDamageFullHealthCheck.cs
Comment (MustaphaTR) [reply OpenRA.Mods.Common/UpdateRules/Rules/20180923/RemoveNegativeDamageFullHealthCheck.cs]: `those` here refers to the weapons. I dunno if this can be worded better, but current sentence is gramatically and meaningly correct, i think.

Comment (reaperrr) [reply OpenRA.Mods.Common/UpdateRules/Rules/20180923/RemoveNegativeDamageFullHealthCheck.cs]: I agree with @abcdefg30. this is not entirely grammatically correct.

"by those grant" should be "by those by granting"...

File: OpenRA.Mods.Common/UpdateRules/Rules/20180923/RemoveNegativeDamageFullHealthCheck.cs
Comment (reaperrr) [inline OpenRA.Mods.Common/UpdateRules/Rules/20180923/RemoveNegativeDamageFullHealthCheck.cs]: ... and here you need to insert a 'to' before the 'remove.

**Comment by pchote** [comment]
Can we get this fixed up and merged?

**Comment by MustaphaTR** [comment]
Updated.

**Review by pchote — DISMISSED** [review]
There were still, IMO, some issues with the English in the update rule. I have pushed over this with revised wording. LGTM now, but i'd prefer if @abcdefg30 or someone else could confirm a :+1: on the new wording before merging.

File: OpenRA.Mods.Common/UpdateRules/Rules/20180923/RemoveNegativeDamageFullHealthCheck.cs
Comment (reaperrr) [inline OpenRA.Mods.Common/UpdateRules/Rules/20180923/RemoveNegativeDamageFullHealthCheck.cs]: This seems to be missing 'weapons' after the 'Negative-damage'.

**Review by reaperrr — APPROVED** [review]

**Pull request merged**
